### DISCUSSION TITLE
GitHub pages & localhost link problem

### DISCUSSION THREAD

**Discussion by JR1811** [sharadcodes/jekyll-theme-serial-programmer#16] [Q&A / Get Help]
Hello,

I started working with Jekyll a few days ago and found this beautiful theme.
When I work with the project on localhost everything is fine.

But when I use it with GitHub Pages the links stop working and the Blog entries don't show up anymore.

Even the 404 Page's "Homepage" link just redirects back to the same link of the link written in the browser URL.
But the big title element at the top of the 404 Page leads back to the homepage with the correct link.

I think there is a problem with the relative links in the markdown files and not with the HTML files but I can't find the issue.
What could be the problem? 😅

**Comment by sharadcodes** [comment]
One common issue with relative links on GitHub Pages is that they may not resolve correctly due to the way the site is served from a subdirectory. You can try using absolute links instead of relative links to ensure they resolve correctly.
Another possible issue could be related to the base URL of your site. Make sure that the baseurl setting in your _config.yml file is set correctly. If your site is being served from a subdirectory, you'll need to include the subdirectory in the baseurl.
You can also try running Jekyll locally with the --baseurl option to see if that resolves the issue. For example, if your site is being served from a subdirectory called my-site, you would run Jekyll locally with jekyll serve --baseurl '/my-site'.
